src/gl/polygonFeature.js

```diff
@@ -0,0 +1,335 @@
+
+    m_this.data().forEach(function (item) {
+      polygon = m_this.polygon()(item, itemIndex);
+      polygonItem = polygon.outer;
+      holes = polygon.inner;
+      polygonItemCoordIndex = 0;
+      extRing = [];
+      extIndex = 0;
+      extLength = polygonItem.length - 1;
+      extRing[0] = [];
+      intIndex = 0;
+
+      polygonItem.forEach(function (extRingCoords) {
+        if (extIndex !== extLength) {
+          //extRing = extRing.concat(extRingCoords);
+          posInstance = posFunc(extRingCoords,
+                                polygonItemCoordIndex,
+                                item, itemIndex);
+          if (posInstance instanceof geo.latlng) {
+            extRing[0].push({x: posInstance.x(), y: posInstance.y()});
+          } else {
+            extRing[0].push({x: posInstance.x, y: posInstance.y});
+          }
+
+          fillColorInstance = fillColorFunc(polygonItemCoords,
+                                            polygonItemCoordIndex,
+                                            item, itemIndex);
+          fillColor.push([fillColorInstance.r,
+                          fillColorInstance.g,
+                          fillColorInstance.b]);
+          fillOpacity.push(fillOpacityFunc(polygonItemCoords,
+                                           polygonItemCoordIndex,
+                                           item,
+                                           itemIndex));
+          polygonItemCoordIndex += 1;
+        }
+        extIndex += 1;
+      });
+
+      polygonItemCoordIndex = 0;
+      holes.forEach(function (hole) {
+        extRing[intIndex + 1] = [];
+        hole.forEach(function (intRingCoords) {
+          posInstance = posFunc(intRingCoords, polygonItemCoordIndex,
+                                item, itemIndex);
+          fillColorInstance = fillColorFunc(intRingCoords,
+                                            polygonItemCoordIndex,
+                                            item, itemIndex);
+          fillColor.push([fillColorInstance.r,
+                          fillColorInstance.g,
+                          fillColorInstance.b]);
+          fillOpacity.push(fillOpacityFunc(intRingCoords,
+                                           polygonItemCoordIndex,
+                                           item, itemIndex));
+          polygonItemCoordIndex += 1;
+          if (posInstance instanceof geo.latlng) {
+            extRing[intIndex + 1].push({x: posInstance.x(), y: posInstance.y()});
+          } else {
+            extRing[intIndex + 1].push({x: posInstance.x, y: posInstance.y});
+          }
+        });
+        intIndex += 1;
+      });
+
+      //console.log("extRing ", extRing);
+      //console.log("result", PolyK.Triangulate(extRing));
+      triangList = triangulator.triangulate_polygon(extRing);
+      newTriangList = [];
+
+      triangList.forEach(function (newIndices) {
+        newTriangList = newTriangList.concat(newIndices);
```

Comment:
I think the following might be faster:
```javascript
Array.prototype.push.apply(newTriangList, newIndices);
```
It extends the array, rather than concatenating into a new array.  Although without preallocation, it is hard to say.